src/lib.rs

```diff
@@ -363,19 +378,21 @@ impl NodeId {
     }
 
     /// Insert a new sibling before this node.
-    pub fn insert_before<T: Sync>(self, new_sibling: NodeId, arena: &mut Arena<T>) {
+    pub fn insert_before<T>(self, new_sibling: NodeId, arena: &mut Arena<T>) {
         new_sibling.detach(arena);
         let previous_sibling_opt;
         let parent_opt;
         {
-            let (self_borrow, new_sibling_borrow) =
-                arena
-                    .nodes
-                    .get_pair_mut(self.index, new_sibling.index, "Can not insert a node before itself");
+            let (self_borrow, new_sibling_borrow) = arena.nodes.get_pair_mut(
```

Comment:
I think the formatting is okay, I just formatted it with 120 chars per line in my rustfmt config.